### PR TITLE
List genes first commit

### PR DESCRIPTION
**Pull request by ogibson**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-458?atlOrigin=eyJpIjoiYTA3ODBhYWUwMmE1NDUzODlhYzIxMzkwYWM2ZjZmNWEiLCJwIjoiaiJ9

#### Link to staging deployment URL 

https://ui-gxc51w7udznz6mygakjcu7tnpa.scp-staging.biomage.net/

#### Links to any Pull Requests related to this

https://github.com/biomage-ltd/ui/pull/121

#### Anything else the reviewers should know about the changes here

These changes won't be reflected on the staging link until the UI pull request is merged. 

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X ] Tested locally with Inframock
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [x] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ X] Photo of a cute animal attached to this PR

image

### CODE REVIEW

File: python/src/tasks/list_genes.py [hunk 1]
```diff
@@ -19,66 +23,33 @@ def _format_result(self, result, total):
...
+        # Convert to dataframe to prepare the data for the UI
+        resR = pd.DataFrame(resR)
         total = 0
-
-        if len(result) > 0:
-            total = result["full_count"][0]
-
+        if len(resR) > 0:
+            total = resR["full_count"][0]
+        resR = resR.drop("full_count", axis=1)
```
Comment (seb-mueller) [inline python/src/tasks/list_genes.py:51]: maybe put the logic into R to have all computing at the same place (use transpose() ?)

File: r/src/list_genes.r [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#
+#
+# req has body with        {'selectFields'= {"gene_names", "dispersions"},
+#                          'orderBy'= {"gene_names", "dispersions"},
+#                          'geneNamesFilter' = string with r placeholders (^ before for endswith and $ after for startswith),
```
Comment (seb-mueller) [inline r/src/list_genes.r:5]: Names = Pattern

**Review by seb-mueller — APPROVED** [review]
- when changing dropdown option it loads quite long (it shouldn't?)

**Comment by marcellp** [comment]
LGTM regarding the Regex sanitation.

**Pull request merged**